core/base/src/TROOT.cxx

```diff
@@ -2069,7 +2073,7 @@ void TROOT::InitInterpreter()
       }
 
       char *libcling = gSystem->DynamicPathName("libCling");
-      gInterpreterLib = dlopen(libcling, RTLD_LAZY|RTLD_LOCAL);
+      gInterpreterLib = dlopen(libcling, RTLD_LAZY|RTLD_LOCAL|RTLD_DEEPBIND);
```

Comment:
In order to test the proposed approach we should remove this line https://github.com/root-project/root/blob/master/interpreter/CMakeLists.txt#L117